base_delivery_carrier_label/models/stock_quant_package.py

```diff
@@ -22,7 +22,7 @@ def _compute_weight(self):
         for pack in self:
             if pack.shipping_weight:
                 pack.weight = pack.shipping_weight
-            elif not pack.quant_ids and not self.env.context.get("picking_id"):
+            elif pack.quant_ids and not self.env.context.get("picking_id"):
```

Comment:
I'm curious about the change in the test here. Why are we changing `if not pack.quant_ids` to `if  pack.quant_ids` and keeping the same code in the block?

Reply:
I think it's the difference between versions because this code is taken from the standard as an example.
```
        for package in self:
            weight = package.package_type_id.base_weight or 0.0
            if self.env.context.get('picking_id'):
                package.weight = weight + package_weights[package.id]
            else:
                for quant in package.quant_ids:
                    weight += quant.quantity * quant.product_id.weight
                package.weight = weight
```
from https://github.com/odoo/odoo/blob/32b26c1a3e0adbf78c0a8f9b69178e14c712fd3a/addons/stock_delivery/models/stock_quant_package.py#L26-L33
I've tried with the tests and we always have `pack.quant_ids` created.
BTW, `delivery` was refactored in v17 and moved to `stock_delivery`.